### PR TITLE
DBZ-5997 Debezium Storage add support for Apache RocketMQ

### PR DESCRIPTION
**Pull request by sunxiaojian**
https://issues.redhat.com/browse/DBZ-5997

### CODE REVIEW

**Comment by jpechane** [comment]
@vjuranek Could you please review this one too? Thanks

File: debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/RocketMqAdminUtil.java
Comment (vjuranek) [inline debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/RocketMqAdminUtil.java:86]: please use `initDefaultMqProducer` instead (and similar in other method names)

File: documentation/modules/ROOT/pages/operations/debezium-server.adoc
Comment (vjuranek) [inline documentation/modules/ROOT/pages/operations/debezium-server.adoc:319]: I'd remove `If using RocketMQ to store schema history, you need to` and use just `The RocketMQ storage topic where history will be stored` or simply use same `withDescription()` strings from `RocketMqSchemaHistory`. Similar to other config items.

File: debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java
Comment (vjuranek) [inline debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java:147]: Instead of `6`, please use `PermName.PERM_READ | PermName.PERM_WRITE` - it's more easy to understand and also use RocketMQ predefined constant instead of hard-coded numbers.

File: debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java
Comment (vjuranek) [inline debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java:199]: this will return without being sure the event was successfully written to the topic - we should wait here for the results if it was written or not, e.g. by `latch.await()`, similar to RockerMQ sink adapter.

File: debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java
Comment (vjuranek) [inline debezium-storage/debezium-storage-rocketmq/src/main/java/io/debezium/storage/rocketmq/history/RocketMqSchemaHistory.java:53]: I'd assume that `UNLIMITED_VALUE` will contain something like infinity, not `-1`. Maybe just use it directly in the code `long lastProcessedOffset = -1`, it's IMHO more understandable when `long lastProcessedOffset = UNLIMITED_VALUE;` (and yes, I noticed that it's used also in `KafkaSchemaHistory`, but it doesn't mean that I like it, it's already there:-()

**Review by vjuranek — CHANGES_REQUESTED** [review]
Overall LGTM, just a few mostly code-style comments. Also `checkstyle` fails - these issues are not mentioned here, just run `mvn clean verify` to see list of issues.

**Comment by sunxiaojian** [comment]
> Overall LGTM, just a few mostly code-style comments. Also `checkstyle` fails - these issues are not mentioned here, just run `mvn clean verify` to see list of issues.

Thanks review PR, I have modified according to your suggestions

**Review by vjuranek — CHANGES_REQUESTED** [review]
@sunxiaojian thanks! There are still few javadoc params without description - please remove them and also there are some checkstyle issues. Please fix them (or check "Allow edits and access to secrets by maintainers " and I can push the changes). Once done, IMHO ready to merge (and once we have working CI again).

**Comment by sunxiaojian** [comment]
> 

ok， I made some fixes

**Comment by sunxiaojian** [comment]
> @sunxiaojian thanks! There are still few javadoc params without description - please remove them and also there are some checkstyle issues. Please fix them (or check "Allow edits and access to secrets by maintainers " and I can push the changes). Once done, IMHO ready to merge (and once we have working CI again).

ok， I made some fixes

**Comment by sunxiaojian** [comment]
> Thanks @sunxiaojian ! LGTM now, will merge once we fix CI and tests pass.

The failure of CI may be a bug (https://github.com/tj-actions/changed-files/issues/419). May need to upgrade the tj-actions/changed-files version?

**Comment by jpechane** [comment]
@sunxiaojian Interesting notice, could you please try to updated the PR with the Actions change so we can see if it is root cause or not?

**Comment by vjuranek** [comment]
@sunxiaojian yes, there's already a [PR for it](https://github.com/debezium/debezium/pull/4230) - you can use it and rebase this PR on top of it to test if it really fixed the issue or not as suggested by @jpechane

**Comment by sunxiaojian** [comment]
> @sunxiaojian yes, there's already a [PR for it](https://github.com/debezium/debezium/pull/4230) - you can use it and rebase this PR on top of it to test if it really fixed the issue or not as suggested by @jpechane

After rebase this PR , the issue is fixed

**Comment by sunxiaojian** [comment]
> @sunxiaojian yes, there's already a [PR for it](https://github.com/debezium/debezium/pull/4230) - you can use it and rebase this PR on top of it to test if it really fixed the issue or not as suggested by @jpechane

After rebase this PR , the issue is fixed

**Comment by vjuranek** [comment]
Merge, thanks @sunxiaojian !

**Pull request merged**
